### ISSUE TITLE
PHP magic hash mode

### ISSUE THREAD

**Issue by bananabr** [hashcat/hashcat#3563]
**Describe the feature**
Include an option for hashcat to report every input whose calculated hash matches the format /^0+e[0-9]+$/. This would allow researchers to look for PHP magic hashes for every algorithm supported by hashcat.

**Current behavior**
There is no way for hashcat to report calculated hashes whose hexadecimal representations match a given pattern.

**Expected behavior**
Hashcat includes an option to report passwords whose calculated hash match a given regex pattern.

**Hashcat version (please complete the following information):**
 - OS: Linux
 - Distribution: Ubuntu 20.04
 - Version: 6.2.6

**Comment by Chick3nman** [comment]
This is very unlikely to be supported officially and while I, and others, have previous done work on modified versions of hashcat for that purpose, we also found that there is little utility to supporting a tool for magic hash generation since you can reuse existing magic hashes without much issue. Finding new ones is generally more of a novelty than anything else. For some examples: https://github.com/spaze/hashes

**Comment by bananabr** [comment]
For my use case, the target application prefix my input with a static string. Therefore, I am not free to use a previously calculated magic hash input.

**Comment by Chick3nman** [comment]
Fair enough, that is indeed a valid use case, though I still don't see this being added to the main project unless someone can find an elegant way to modify the comparison logic and restrict it's use to only a few select kernels(not all make use of the same comparison code). I had originally looked into a fork with a few roughed out modes but even that seemed like more work than necessary when a quick kernel mod for your target mode is likely all that's needed and it can be done with the latest hashcat at the time of need to take advantage of the newest code base. This sort of matching is actually not super compatible with how hashcat currently organizes and marks targets and the overhead would likely be pretty big if added to all current modes(new kernel per mode or changes to all existing kernels i believe). If you compare the operation of a kernel that's been hacked for this purpose to normal operation, you can see what I mean. I'd normally field requests like this as one-off asks elsewhere because of that, happy to help solve this that way if possible.

**Comment by bananabr** [comment]
Is there documentation on how to write custom kernels?

**Comment by Chick3nman** [comment]
There is some documentation here: https://github.com/hashcat/hashcat/blob/master/docs/hashcat-plugin-development-guide.md

**Comment by Chick3nman** [comment]
If you need help with getting a quick implementation of this done, let me know. Like i said before, doing this is a bit of a hack and the dev guide probably will take longer than just using a pattern that i've already used before.

**Comment by philsmd** [comment]
Couldn't you just `grep` your outfile (-o) or pot files with `grep` ? or, if you only want to focus on this pattern, just run that particular mask ?

**Comment by Chick3nman** [comment]
> Couldn't you just `grep` your outfile (-o) or pot files with `grep` ? or, if you only want to focus on this pattern, just run that particular mask ?

The intended target is any (unknown) plaintext that produces any hash matching a specific format. This is similar to searching for a substring of  a hash or for a "vanity" hash. Magic hashes must start with "0e" or "00e" and only contain 0-9 in the rest of the hash. This is basically inverse of how hashcat normally works since we don't have a hash to target at the start but modifying the kernel with a static comparison/mark_hash to match any hash with this format is possible, though it's a hack and abuses/bypasses a lot of the normal operation.
